genesyscloud/outbound_filespecificationtemplate/resource_genesyscloud_outbound_filespecificationtemplate_utils.go

```diff
@@ -0,0 +1,142 @@
+package outbound_filespecificationtemplate
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v121/platformclientv2"
+)
+
+func buildSdkOutboundFileSpecificationTemplate(d *schema.ResourceData) platformclientv2.Filespecificationtemplate {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	format := d.Get("format").(string)
+	numberOfHeaderLinesSkipped := d.Get("number_of_header_lines_skipped").(int)
+	numberOfTrailerLinesSkipped := d.Get("number_of_trailer_lines_skipped").(int)
+	header := d.Get("header").(bool)
+	delimiter := d.Get("delimiter").(string)
+	delimiterValue := d.Get("delimiter_value").(string)
+
+	sdkFileSpecificationTemplate := platformclientv2.Filespecificationtemplate{
+		NumberOfHeadingLinesSkipped:  &numberOfHeaderLinesSkipped,
+		NumberOfTrailingLinesSkipped: &numberOfTrailerLinesSkipped,
+		Header:                       &header,
+		ColumnInformation:            buildSdkOutboundFileSpecificationTemplateColumnInformationSlice(d.Get("column_information").([]interface{})),
+		PreprocessingRules:           buildSdkOutboundFileSpecificationTemplatePreprocessingRulesSlice(d.Get("preprocessing_rule").([]interface{})),
+	}
+
+	if name != "" {
+		sdkFileSpecificationTemplate.Name = &name
+	}
+	if description != "" {
+		sdkFileSpecificationTemplate.Description = &description
+	}
+	if format != "" {
+		sdkFileSpecificationTemplate.Format = &format
+	}
+	if delimiter != "" {
+		sdkFileSpecificationTemplate.Delimiter = &delimiter
+	}
+	sdkFileSpecificationTemplate.DelimiterValue = &delimiterValue
+	return sdkFileSpecificationTemplate
+}
+
+func buildSdkOutboundFileSpecificationTemplateColumnInformationSlice(columnInformation []interface{}) *[]platformclientv2.Column {
+	if columnInformation == nil || len(columnInformation) < 1 {
+		return nil
+	}
+	sdkColumnInformationSlice := make([]platformclientv2.Column, 0)
+	for _, columnInfo := range columnInformation {
+		if columnInfoMap, ok := columnInfo.(map[string]interface{}); ok {
+			var sdkColumnInformation platformclientv2.Column
+
+			if columnNameStr, ok := columnInfoMap["column_name"].(string); ok {
```

Comment:
Similar to the read function we have functions in the `resourcedata` package that can be used in the build functions to reduce the number of if statements. If you don't know which to use you can see example in the other resources that have been put into their own package.